### ISSUE TITLE
Auto-connect to remote on startup

### ISSUE THREAD

**Issue by peterkappelt** [Microsoft/vscode-remote-release#120]
Hi everyone,

awesome feature you are working on here!

I think it might be useful to add a command line parameter that allows to automatically connect to a remote host when VS Code starts, maybe something like: `--connect-remote devserver01`.

One could create shortcuts to automatically connect with this feature

**Comment by shivapoudel** [comment]
@bamurtaugh Yeah, similar approach but slightly different. I was talking about workspace setting like this:

```
{
	"folders": [
		{
			"path": "D:\\uppy-companion"
		}
	],
	"settings": {
		"wsl": "<distro>|bool"
	}
}
```

**Comment by GuilhermeWerner** [comment]
> @bamurtaugh Yeah, similar approach but slightly different. I was talking about workspace setting like this:
> 
> ```
> {
> 	"folders": [
> 		{
> 			"path": "D:\\uppy-companion"
> 		}
> 	],
> 	"settings": {
> 		"wsl": "<distro>|bool"
> 	}
> }
> ```

Exactly what I was thinking, I was going to open this request, but I saw this one.

That would be a very welcome feature, in my opinion, it would be very practical, and it would help a lot of people.

**Comment by lohgun** [comment]
Hi!

I'm looking for the same feature, but I was thinking of something like this (more similar to an ssh command):

```bash
code [--remote] [<user>@]<host>:path/to/prj
```

Following above examples:

```bash
# absolute
code --remote devserver01:/d/uppy-companion

# relative to home
code --remote devserver01:src/project2
```

**Comment by olliefr** [comment]
I'm surprised that this option doesn't exist yet 🙃 +1 for the request. I work in W10+WSL2(Ubuntu). I don't actually compile/run anything outside of WSL2 Remote.

**Comment by aviadgal** [comment]
> I'm surprised that this option doesn't exist yet 🙃 +1 for the request. I work in W10+WSL2(Ubuntu). I don't actually compile/run anything outside of WSL2 Remote.

+1

**Comment by GuilhermeWerner** [comment]
> I'm surprised that this option doesn't exist yet 🙃 +1 for the request. I work in W10+WSL2(Ubuntu). I don't actually compile/run anything outside of WSL2 Remote.

+1

**Comment by iamjohnnyli** [comment]
> I'm surprised that this option doesn't exist yet 🙃 +1 for the request. I work in W10+WSL2(Ubuntu). I don't actually compile/run anything outside of WSL2 Remote.

+1

**Comment by hendjoshsr71** [comment]
> I'm surprised that this option doesn't exist yet 🙃 +1 for the request. I work in W10+WSL2(Ubuntu). I don't actually compile/run anything outside of WSL2 Remote.

+1

**Comment by diegorochawebdg** [comment]
> I'm surprised that this option doesn't exist yet 🙃 +1 for the request. I work in W10+WSL2(Ubuntu). I don't actually compile/run anything outside of WSL2 Remote.

+1

**Comment by steenda** [comment]
> I'm surprised that this option doesn't exist yet 🙃 +1 for the request. I work in W10+WSL2(Ubuntu). I don't actually compile/run anything outside of WSL2 Remote.

+1

**Comment by steenda** [comment]
Just found an easy workaround for WSL. Just create a shortcut for 'wsl code' and VSCode will open in WSL :)

**Comment by MuhammadAlBarham** [comment]
I think this is a great feature. Using Windows 10 and WSL2 (Ubuntu) is so important.

Please, add this feature to make our life easier 💯

**Comment by vuvgaG** [comment]
Yes need to connect to a remote via ssh

**Comment by KubaJastrz** [comment]
> Just found an easy workaround for WSL. Just create a shortcut for 'wsl code' and VSCode will open in WSL :)

Yeah, I created a `.bat` file:
```
wsl code
```

But to make sure that the Remote will be always opened, I had to also disable automatic reopening of closed windows:

```json
// settings.json
{
  "window.restoreWindows": "none"
}
```

**Comment by kusznerr** [comment]
Team - how many upvotes we need to give for the proposed feature to be implemented. ? I saw multiple other threads being closed and referenced to this one. As you can see a dev community would really welcome this particular feature to be enabled 👍

**Comment by csharpdf** [comment]
I'd really appreciate a feature for Remote - WSL that does exactly this.

**Comment by saymondamasio** [comment]
is there any update on this?

**Comment by migom6** [comment]
Was wondering if anyone have found a way to launch vscode connected to a remote machine automatically. Something along the lines of deeplink or workspace settings as an input to 
`code --workspace workspace-file`

**Comment by kobenauf** [comment]
A slightly different take still, I would love if there were a flag like -c that opened the specified folder with whatever .devcontainer identifies if it exists, e.g.
`code -c .`

**Comment by NFSS10** [comment]
I would like a user setting that would let us do this

**Comment by naturedamends** [comment]
I have this which works. 

```
$ cat /etc/hosts
## ..

192.168.0.162  host.example.com
```
```
$ cat ~/.ssh/config
Host primary
	HostName host.example.com
	User ubuntu
```
```
$ cat primary.code-workspace
{
	"folders": [
		{
			"uri": "vscode-remote://ssh-remote+primary/home/ubuntu/repos/workshop/dotfiles"
		}
	],
	"remoteAuthority": "ssh-remote+primary",
	"settings": {
		"terminal.integrated.env.linux": {
			"CLR_OPENSSL_VERSION_OVERRIDE": "1.1"
		}
	},
	""
}
```
